### ISSUE TITLE
Breaking the pyqg dependency

### ISSUE THREAD

**Issue by jdenholm** [m2lines/equation_discovery#5]
The ``utils.py`` currently depends on the [``pyqg``](https://github.com/pyqg/pyqg) repository, but Laure says she would like this equation discovery repository to be completely independent of [``pyqg``](https://github.com/pyqg/pyqg) .

- Is dropping the [``pyqg``](https://github.com/pyqg/pyqg)  dependency definitely what we want to achieve?
- How can we refactor/redesign this repository such that it is standalone with no [``pyqg``](https://github.com/pyqg/pyqg)  dependency?

**Comment by LaureZanna** [comment]
Yes, it would be great to remove` pyqg ` dependency so that users can feed in their data without having to install` pyqg` which is a numerical solver.

**Comment by asross** [comment]
My take on this is that we should support `pyqg` and require it for certain things (e.g. computing similarity metrics that are specifically designed for `pyqg.Model`s), but we shouldn't require it for the entire library, since much of the code (e.g. the actual hybrid symbolic procedure) can operate just as easily on an `xarray.Dataset`.

To accomplish that, I think we'll need to do something like:
- try `import`ing `pyqg` and store a boolean indicating whether the import succeeded
- replace existing type checks for `pyqg.Model` with something like `is_pyqg_model` that returns `isinstance(pyqg.Model)` if the import succeeded and `False` if the import failed

Does that make sense? I think that's a better approach than removing all references to `pyqg` altogether, since then it will become very difficult to run the library with pyqg models.

**Comment by LaureZanna** [comment]
Thanks @asross; i like the idea. I feel as long as one can use the hybrid symbolic procedure without `pyqg`  that would be really great (including calculating the derivatives).
